src/main/cli/dirs.ts

```diff
@@ -130,19 +117,22 @@ export function getTorBinary(): Binary {
 
 export async function getFileData(file: string): Promise<string> {
   try {
-    const prevLogData = await fs.readFile(file, {
+    return await fs.readFile(file, {
       encoding: 'utf8',
     });
-    return prevLogData;
   } catch (e) {
     throw new Error(`Failed to read file! Path: ${file} Error: ${e}`);
   }
 }
 
-export async function getCliLogStdOut(swapId: string): Promise<string> {
-  const logFile = await getCliLogFile(swapId);
-  const logData = await getFileData(logFile);
-  return logData;
+export default async function getSavedLogsOfSwapId(
+  swapId: string
+): Promise<CliLog[]> {
+  const logsFile = await getCliLogFile();
+  const fileData = await getFileData(logsFile);
+  const allLogs = getLogsFromRawFileString(fileData);
+
+  return allLogs.filter((log) => getCliLogSpanSwapId(log) === swapId);
 }
```

Comment:
**Suggestion:** Consider adding error handling for the async function 'getSavedLogsOfSwapId' at line 128. Currently, if an error occurs during the execution of 'getCliLogFile' or 'getFileData', it will not be caught and could lead to unexpected behavior.
```suggestion
export default async function getSavedLogsOfSwapId(swapId: string): Promise<CliLog[]> {
  try {
    const logsFile = await getCliLogFile();
    const fileData = await getFileData(logsFile);
    ...
  } catch (e) {
    logger.error(`Error occurred: ${e.message}`);
  }
}
```